### PR TITLE
fix(tf): dispatch --init-model to checkpoint pre-inspection

### LINKED ISSUE #5679: [Code scan] Inspect TensorFlow --init-model checkpoints before graph build
This issue comes from a Codex global scan of `deepmodeling/deepmd-kit` at commit `73de44b1f94471b2e3bdb6b11f57b34d7bc791bb`.

## Problem

`RunOptions` records `--init-model` mode as `"init_from_model"`:

https://github.com/deepmodeling/deepmd-kit/blob/73de44b1f94471b2e3bdb6b11f57b34d7bc791bb/deepmd/tf/train/run_options.py#L146-L152

But `DPTrainer.build()` checks for `"init_model"` instead:

https://github.com/deepmodeling/deepmd-kit/blob/73de44b1f94471b2e3bdb6b11f57b34d7bc791bb/deepmd/tf/train/trainer.py#L257-L264

As a result, `_init_from_ckpt()` is called for `restart`, but not for `--init-model`:

https://github.com/deepmodeling/deepmd-kit/blob/73de44b1f94471b2e3bdb6b11f57b34d7bc791bb/deepmd/tf/train/trainer.py#L906-L918

That pre-inspection is the code path that detects compressed checkpoint metadata and sets `self.ckpt_meta` before graph construction. The graph build later passes `ckpt_meta=self.ckpt_meta` into the model:

https://github.com/deepmodeling/deepmd-kit/blob/73de44b1f94471b2e3bdb6b11f57b34d7bc791bb/deepmd/tf/train/trainer.py#L334-L343

## Impact

TensorFlow `dp train --init-model ...` can skip checkpoint pre-inspection entirely. Compressed-checkpoint initialization is especially exposed because the metadata that should be discovered by `_init_from_ckpt()` may never be set before `model.build(...)`.

## Suggested fix

Change the trainer branch to check `self.run_opt.init_mode == "init_from_model"`, or use a shared constant/enum for the mode strings used by `RunOptions` and `DPTrainer`.

### PR DESCRIPTION
**Pull request by wanghan-iapcm**
## Problem

Fixes #5679. `RunOptions` records `dp train --init-model` as `init_mode == "init_from_model"` (`deepmd/tf/train/run_options.py`), but `DPTrainer.build()` dispatched the init step on the literal `"init_model"`. The two strings never matched, so `_init_from_ckpt(...)` was skipped for `--init-model`. That pre-inspection is the step that imports the source checkpoint's meta graph and, when the checkpoint is a `compressed_model`, sets `self.ckpt_meta` before the graph is built (the graph build later consumes `ckpt_meta`). With the mismatch, a compressed-checkpoint `--init-model` run builds the graph without its checkpoint metadata.

The bug was masked for the common case because uncompressed `--init-model` still works: the variables are restored later in `_init_session`, which uses the correct `"init_from_model"` literal and does not need `ckpt_meta`. So only compressed-checkpoint initialization was actually exposed, and no test exercised it.

## Fix

Correct the dispatch literal to `"init_from_model"`. To make the dispatch unit-testable — the reason the mismatch went uncaught is that it lived inline in the heavyweight `build()` — the four-way init dispatch is extracted into a small `_init_from_run_opt()` helper. The restart, init-from-frozen-model, and finetune branches already used the correct literals and are unchanged in behavior.

## Test

Adds `source/tests/tf/test_trainer_init_mode.py`, which drives `_init_from_run_opt` on a stub trainer with the three concrete initializers mocked and asserts each `init_mode` routes correctly. On the old literal, `init_from_model` routes to nothing (the test fails); with the fix it reaches `_init_from_ckpt(init_model)`. The test also covers `restart`, `init_from_frz_model`, `finetune`, and the scratch no-op. This dispatch previously had no coverage.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved training startup by centralizing initialization dispatch, ensuring the correct checkpoint/frozen-model/pretrained initialization path is selected for restore, restart, fine-tuning, and frozen-model workflows.
  * If training is configured for “scratch” (or an unsupported init mode), initialization pre-inspection steps are skipped and startup proceeds.

* **Tests**
  * Added regression tests covering each supported initialization path to prevent future startup regressions.

### CODE REVIEW

**Review by njzjz — APPROVED** [review]

File: source/tests/tf/test_trainer_init_mode.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+``RunOptions`` records ``--init-model`` as ``init_mode == "init_from_model"``,
+but ``DPTrainer`` used to dispatch on the literal ``"init_model"``.  The
+mismatch silently skipped ``_init_from_ckpt`` for ``--init-model``, so the
+compressed-checkpoint pre-inspection that sets ``ckpt_meta`` before graph
+construction never ran.  This checks that each init mode routes to the right
+initializer, using the mode strings ``RunOptions`` actually produces.
+"""
+
+import types
+import unittest
```
[the remark this answers is not in the document]
Comment (wanghan-iapcm) [reply source/tests/tf/test_trainer_init_mode.py]: Fixed: consolidated into a single `from unittest import (TestCase, main, mock)`.
